packages/p2p/src/components/my-ads/form-ads-wrapper.jsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { MobileFullPageModal } from '@deriv/components';
+import { isMobile } from '@deriv/shared';
+import { observer } from 'mobx-react-lite';
+import { localize } from 'Components/i18next';
+import { useStores } from 'Stores';
+
+const FormAdsWrapper = ({ children }) => {
+    const { my_ads_store } = useStores();
+
+    return isMobile() ? (
+        <MobileFullPageModal
+            className='p2p-my-ads__form'
+            is_modal_open={my_ads_store.show_ad_form}
+            page_header_text={localize('Create new ad')}
+            pageHeaderReturnFn={() => my_ads_store.setShowAdForm(false)}
+            is_flex
+        >
+            {children}
+        </MobileFullPageModal>
+    ) : (
+        <React.Fragment>{children}</React.Fragment>
+    );
+};
+
+FormAdsWrapper.propTypes = {
+    children: PropTypes.any,
+    is_modal_open: PropTypes.bool,
```

Comment:
I'm retrieving it from the store and that's why it needs to be listed as a prop.

Reply:
Why it needs to be listed as FormAdsWrapper's propTypes? This component does not accept such prop.

Reply:
Hi @reza-fs, this is something we wanted to change to an inject/connect function and hence kept all of these as propTypes in all P2P components but since now we realize that's a messy and not a good solution, we'll be removing all of them to clean up all unnecessary propTypes. Thanks! :)